src/zabapgit_popups.prog.abap

```diff
@@ -313,48 +310,69 @@ CLASS lcl_popups IMPLEMENTATION.
 
   METHOD create_tag_popup.
 
-    DATA: lv_answer TYPE c LENGTH 1,
-          lt_fields TYPE TABLE OF sval.
+    DATA: lv_answer          TYPE c LENGTH 1,
+          lt_fields          TYPE TABLE OF sval,
+          lv_exit_while_loop TYPE abap_bool.
 
     FIELD-SYMBOLS: <ls_field> LIKE LINE OF lt_fields.
 
-    CLEAR: ev_name, ev_cancel.
+    CLEAR: ev_name, ev_cancel, ev_sha1.
 
-    add_field( EXPORTING iv_tabname    = 'TOAVALUE'
-                         iv_fieldname  = 'REFER_CODE'
+    add_field( EXPORTING iv_tabname    = 'TBDTPPT'
+                         iv_fieldname  = 'P_TEXT'
                          iv_fieldtext  = 'SHA'
-                         iv_value      = iv_sha1(7)
-                         iv_field_attr = '05'
+                         iv_value      = iv_sha1
+                         iv_obligatory = abap_true
                CHANGING ct_fields      = lt_fields ).
 
-    add_field( EXPORTING iv_tabname   = 'TEXTL'
-                         iv_fieldname = 'LINE'
-                         iv_fieldtext = 'Name'
-                         iv_value     = 'new-tag-name'
-               CHANGING ct_fields     = lt_fields ).
+    add_field( EXPORTING iv_tabname    = 'TEXTL'
+                         iv_fieldname  = 'LINE'
+                         iv_fieldtext  = 'Name'
+                         iv_obligatory = abap_true
+               CHANGING ct_fields      = lt_fields ).
 
-    CALL FUNCTION 'POPUP_GET_VALUES'
-      EXPORTING
-        popup_title     = 'Create tag'
-      IMPORTING
-        returncode      = lv_answer
-      TABLES
-        fields          = lt_fields
-      EXCEPTIONS
-        error_in_fields = 1
-        OTHERS          = 2 ##NO_TEXT.
-    IF sy-subrc <> 0.
-      zcx_abapgit_exception=>raise( 'error from POPUP_GET_VALUES' ).
-    ENDIF.
+    WHILE lv_exit_while_loop = abap_false.
```

Comment:
looks like this could also be done with DO. ENDO. and removing variable lv_exit_while_loop, but it is just a minor optimization

Reply:
You're right, I barely use DO. ENDO... ;-)